src/blr/cli/init.py

```diff
@@ -0,0 +1,93 @@
+"""
+Create and initialize a new analysis directory.
+"""
+import logging
+import os
+import os.path
+import sys
+from pathlib import Path
+from importlib_resources import read_binary
+
+from ..utils import guess_paired_path
+
+logger = logging.getLogger(__name__)
+
+
+CONFIGURATION_FILE_NAME = "blr.yaml"
+
+
+def add_arguments(parser):
+    parser.add_argument(
+        "--reads1",
+        "--r1",
+        type=Path,
+        metavar="READS",
+        help="First paired-end read file (.fastq.gz). The second is found automatically.",
+    )
+    parser.add_argument("directory", type=Path, help="New analysis directory to create")
+
+
+def main(args):
+    if args.reads1 is None:
+        logger.error("--reads1 must be given")
+        sys.exit(1)
+    init(args.directory, args.reads1)
+
+
+def init(directory: Path, reads1: Path):
+    if " " in str(directory):
+        logger.error("The name of the analysis directory must not contain spaces")
+        sys.exit(1)
+
+    if directory.exists():
+        logger.error(f"The target directory {directory!r} already exists.")
```

Comment:
Should not the script exit here if the directory exists? Otherwise it will just continue until it tries to create the directory in the `create_and_populate_analysis_directory` function and raise an error there.

Reply:
You’re right; the `sys.exit(1)` apparently got lost after I copied the code over from IgDiscover.